src/platforms/esp32/components/avm_builtins/gpio_driver.c

```diff
@@ -348,7 +348,10 @@ static term gpiodriver_set_int(Context *ctx, Context *target, term cmd)
     gpio_set_direction(gpio_num, GPIO_MODE_INPUT);
     gpio_set_intr_type(gpio_num, interrupt_type);
 
-    gpio_isr_handler_add(gpio_num, gpio_isr_handler, data);
+    esp_err_t ret = gpio_isr_handler_add(gpio_num, gpio_isr_handler, data);
+    if (ret != ESP_OK) {
```

Comment:
we try to help branch prediction when handling errors (which are less likely) with UNLIKELY.

Reply:
Yes, thank you. I should have of that, I had already noticed the use of UNLIKELY in many other similar situations.

Reply:
I pushed the requested changes. The push passed all of the test in my branch, I see that it failed the `qemu esp32-test` here, I believe it will pass if the test is re-run.